3rd party tests do not show commands, use "set -x"

So the failure looks like this https://github.com/lfortran/lfortran/actions/runs/16581392048/job/46958529519:
```
Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at c915a24 Compile test_chat with --realloc-lhs
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
curl: (22) The requested URL returned error: 503
Error: Process completed with exit code 22.
```
It doesn't show the command that caused it.

The solution is to use `set -x` in the Bash script. We used to do that, but then we stopped for some reason. Let's put it back in.

--- a/ci/test_third_party_codes.sh
+++ b/ci/test_third_party_codes.sh
@@ -1,5 +1,5 @@
 #!/bin/bash
-set -e  # Exit immediately on any error
+set -ex  # Exit immediately on any error
 
 # Default to gfortran if FC is not set
 : "${FC:=gfortran}"

